runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -264,6 +259,14 @@ public static Environment createProcessEnvironment(
                   .build()
                   .toByteString());
         }
+        if (stagedName == null) {
+          stagedName = createStagingFileName(file, hashCode);
```

Comment:
Where are we setting the hashCoder for this path ? Also are we missing a unit test for this ?

Reply:
I believe existing unit test covers this path. This is the path when a given path doesn't have special '=' character. In that case, we generate the staged name.